Source/WebCore/fileapi/FileReader.cpp

```diff
@@ -150,14 +150,14 @@ void FileReader::abort()
 
 void FileReader::didStartLoading()
 {
-    enqueueTask([this] {
+    enqueueTask([this, protectedThis = Ref { *this }] {
```

Comment:
I think enqueueTask keeps `this` alive as well so this is not entirely satisfactory. Is there a better way?

Reply:
The way we have done this in the past is that we do not capture `this` here. Instead, we have the `enqueueTask()` function pass `this` as parameter to the lambda, since it is the one that protects `this`.